docs/user_authentication/third_party_token_management.md

```diff
@@ -1,27 +1,25 @@
 
 ```sequence {theme: 'simple'}
 Title: Third party idp token management
-Pbench-Dashboard->Web-Server:[1] GET login page
-Web-Server->Pbench-Dashboard:[2] 200 login page
+Pbench-Dashboard->Pbench-Server:[1] GET login page
+Pbench-Server->Pbench-Dashboard:[2] 200 login page
 Pbench-Dashboard->Pbench-Dashboard:[3] Generate Authorization request
-Pbench-Dashboard->Web-Server:[4] 302 Redirect to identity provider
-Web-Server->Identity-Provider:[5] POST Authorization request
-Identity-Provider->Web-Server:[6] 200 Authenticate User
-Identity-Provider->Web-Server:[7] GET consent request
-Web-Server->Identity-Provider:[8] POST consent request payload
-Identity-Provider->Pbench-Dashboard:[9] 200 Authorization code
-Pbench-Dashboard->Identity-Provider:[10] POST Token request
-Identity-Provider->Pbench-Dashboard:[11] 200 Access/Refresh Token payload
-Pbench-Dashboard->Pbench-Server:[12] Send Access_token to get Pbench token
-Pbench-Server->Identity-Provider:[13] GET public Keys
-Identity-Provider->Pbench-Server:[14] 200 Public key
-Pbench-Server->Pbench-Server:[15] Validate the access token
-Pbench-Server->Identity-Provider:[16] [Optional] POST userinfo endpoint
-Pbench-Server->Pbench-Keycloak-server:[17] Exchange idp access token with Pbench token
-Pbench-Keycloak-server->Pbench-Server:[18] 200 new Pbench access_token and refresh_token
-Pbench-Server->Pbench-Dashboard:[19] 200 New Pbench token payload
-Pbench-Dashboard->Pbench-Server:[20] Request access to preotected endpoint
-Pbench-Server->Pbench-Server:[21] Validate Pbench token
-Pbench-Server->Pbench-Server:[22] Role extraction from token
-Pbench-Server->Pbench-Dashboard:[23] 200/401
+Pbench-Dashboard->Identity-Provider:[4] POST Authorization request
+Identity-Provider->Identity-Provider:[5] 200 Authenticate User
+Identity-Provider->Identity-Provider:[6] GET consent
+Identity-Provider->Pbench-Dashboard:[7] 200 Authorization code
+Pbench-Dashboard->Identity-Provider:[8] POST Token request
+Identity-Provider->Pbench-Dashboard:[9] 200 Access/Refresh Token payload
```

Comment:
I have tried to make it less muddled now :)

Reply:
I'm not sure that you've entirely succeeded.  🙁

I suspect that "Pbench Dashboard" may be the wrong label for the third entity.  I suspect that it should be "Browser" or "User Agent" or some such.

Perhaps it might be good to use color or some other form of labeling to indicate which software is running in or otherwise directing the Browser.  Initially (or after Step `#2`...), I expect it's the Pbench Dashboard; however, during Steps `#3` and `#4` it's the IDP's web page.  (How do we get "back" to the Pbench Dashboard after the user completes the authentication with the IDP?  Does the authentication take place in a separate window?  How/why doesn't the Dashboard's authorization request time out if the user takes a long time to provide their credentials?)